msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/GenericAuthority.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class GenericAuthority extends Authority {
+    static final String AUTHORIZATION_ENDPOINT = "oauth2/v2.0/authorize";
```

Comment:
This is not correct. You have to make a call to oidc discovery endpoint (authority + ".well-known/openid-configuration"), and read the authorization endpoint and token endpoint from there. this call should be cached statically (and maybe unit tests need to reset this cache).

Pls see .NET implementation.

Reply:
In my original implementation I saw that the endpoints in the ID labs test environment followed the same "oauth2/v2.0/authorize" we used elsewhere and assumed that was part of the OIDC standard we could expect.

In the latest commit the endpoints are now only set through whatever is found during instance discovery at the ".well-known/openid-configuration" endpoint.